settings.php

```diff
@@ -38,6 +38,13 @@ function ecs($configoption, $params = null) {
     $settings->add(new admin_setting_configtext('equella_url', ecs('url.title'), ecs('url.desc'), ''));
     $settings->add(new admin_setting_configtext('equella_action', ecs('action.title'), ecs('action.desc'), ''));
 
+    $userfieldoptions = array();
+    $userfieldoptions = array('default' => ecs('userfield.username'));
+    foreach($DB->get_records('user_info_field') as $params){
+        $userfieldoptions[$params->shortname] = $params->name;
+    }
+    $settings->add(new admin_setting_configselect('equella_userfield', ecs('userfield.title'), ecs('userfield.desc'), 'default', $userfieldoptions));
```

Comment:
I am not sure if this setting should be under `General settings` which is more focused on oEQ.

I think it's good to have a separate section named `SSO identification` at the bottom of this page.

Reply:
I think this is up to you. I put it under equella_url and equella_action that are use for the equella connection. You still have equella_admin_username in the general settings as well.